Error during build if there's a space in path

### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When building WC according to [these instructions](https://github.com/woocommerce/woocommerce/wiki/How-to-set-up-WooCommerce-development-environment#install-dependencies-and-generate-assets), I get an error during `pnpm install` step if I run it from a folder that has a space in the path (such as `/Users/myname/Local Sites/wctest/...`).

### Expected behavior

No error.

### Actual behavior

```
node_modules/.pnpm/core-js-pure@3.19.1/node_modules/core-js-pure: Running postinstall script, done in 44ms
node_modules/.pnpm/core-js@2.6.12/node_modules/core-js: Running postinstall script, done in 44ms
node_modules/.pnpm/core-js@3.19.1/node_modules/core-js: Running postinstall script, done in 37ms
node_modules/.pnpm/deasync@0.1.21/node_modules/deasync: Running install script, failed in 447ms
.../deasync@0.1.21/node_modules/deasync install$ node ./build.js
│ gyp info it worked if it ends with ok
│ gyp info using node-gyp@8.4.1
│ gyp info using node@16.14.0 | darwin | arm64
│ gyp info find Python using Python version 3.9.10 found at "/opt/homebrew/opt/python@3.9/bin/python3.9"
│ gyp info spawn /opt/homebrew/opt/python@3.9/bin/python3.9
│ gyp info spawn args [
│ gyp info spawn args   '/Users/peter/.nvm/versions/node/v16.14.0/pnpm-global/5/node_modules/.pnpm/pnpm@6.30.1/node_modules/pnpm/dist/node_modules/node-gyp/g
│ gyp info spawn args   'binding.gyp',
│ gyp info spawn args   '-f',
│ gyp info spawn args   'make',
│ gyp info spawn args   '-I',
│ gyp info spawn args   '/Users/peter/Local Sites/wctest/app/woocommerce/node_modules/.pnpm/deasync@0.1.21/node_modules/deasync/build/config.gypi',
│ gyp info spawn args   '-I',
│ gyp info spawn args   '/Users/peter/.nvm/versions/node/v16.14.0/pnpm-global/5/node_modules/.pnpm/pnpm@6.30.1/node_modules/pnpm/dist/node_modules/node-gyp/a
│ gyp info spawn args   '-I',
│ gyp info spawn args   '/Users/peter/Library/Caches/node-gyp/16.14.0/include/node/common.gypi',
│ gyp info spawn args   '-Dlibrary=shared_library',
│ gyp info spawn args   '-Dvisibility=default',
│ gyp info spawn args   '-Dnode_root_dir=/Users/peter/Library/Caches/node-gyp/16.14.0',
│ gyp info spawn args   '-Dnode_gyp_dir=/Users/peter/.nvm/versions/node/v16.14.0/pnpm-global/5/node_modules/.pnpm/pnpm@6.30.1/node_modules/pnpm/dist/node_mod
│ gyp info spawn args   '-Dnode_lib_file=/Users/peter/Library/Caches/node-gyp/16.14.0/<(target_arch)/node.lib',
│ gyp info spawn args   '-Dmodule_root_dir=/Users/peter/Local Sites/wctest/app/woocommerce/node_modules/.pnpm/deasync@0.1.21/node_modules/deasync',
│ gyp info spawn args   '-Dnode_engine=v8',
│ gyp info spawn args   '--depth=.',
│ gyp info spawn args   '--no-parallel',
│ gyp info spawn args   '--generator-output',
│ gyp info spawn args   'build',
│ gyp info spawn args   '-Goutput_dir=.'
│ gyp info spawn args ]
│ gyp info spawn make
│ gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
│   CXX(target) Release/obj.target/deasync/src/deasync.o
│ clang: error: no such file or directory: 'Sites/wctest/app/woocommerce/node_modules/.pnpm/node-addon-api@1.7.2/node_modules/node-addon-api'
│ make: *** [Release/obj.target/deasync/src/deasync.o] Error 1
│ gyp ERR! build error
│ gyp ERR! stack Error: `make` failed with exit code: 2
│ gyp ERR! stack     at ChildProcess.onExit (/Users/peter/.nvm/versions/node/v16.14.0/pnpm-global/5/node_modules/.pnpm/pnpm@6.30.1/node_modules/pnpm/dist/nod
│ gyp ERR! stack     at ChildProcess.emit (node:events:520:28)
│ gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
│ gyp ERR! System Darwin 21.2.0
│ gyp ERR! command "/Users/peter/.nvm/versions/node/v16.14.0/bin/node" "/Users/peter/.nvm/versions/node/v16.14.0/pnpm-global/5/node_modules/.pnpm/pnpm@6.30.1
│ gyp ERR! cwd /Users/peter/Local Sites/wctest/app/woocommerce/node_modules/.pnpm/deasync@0.1.21/node_modules/deasync
│ gyp ERR! node -v v16.14.0
│ gyp ERR! node-gyp -v v8.4.1
│ gyp ERR! not ok
│ Build failed
└─ Failed in 447ms
node_modules/.pnpm/@parcel+watcher@2.0.4/node_modules/@parcel/watcher: Running install script, done in 398ms
node_modules/.pnpm/fsevents@1.2.13/node_modules/fsevents: Running install script...
node_modules/.pnpm/husky@4.3.0/node_modules/husky: Running install script, done in 109ms
node_modules/.pnpm/node-pty@0.9.0/node_modules/node-pty: Running install script...
node_modules/.pnpm/husky@4.3.0/node_modules/husky: Running postinstall script, done in 80ms
node_modules/.pnpm/puppeteer-core@1.12.2/node_modules/puppeteer-core: Running install script, done in 31ms
node_modules/.pnpm/puppeteer@2.1.1/node_modules/puppeteer: Running install script...
 ELIFECYCLE  Command failed with exit code 1.
```

### Steps to reproduce

1. Check out the repo to a folder that has a space in its name
2. Run the build
3. You should see the same error. Maybe it's coming from `deasync` package? Not sure. 

### WordPress Environment

Not relevant.

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

--- a/plugins/woocommerce/changelog/dev-remove-deasync-dep
+++ b/plugins/woocommerce/changelog/dev-remove-deasync-dep
@@ -1,0 +1,4 @@
+Significance: patch
+Type: dev
+
+remove unused deasync dependency

--- a/plugins/woocommerce/package.json
+++ b/plugins/woocommerce/package.json
@@ -79,7 +79,6 @@
 		"chai-as-promised": "7.1.1",
 		"config": "3.3.7",
 		"cross-env": "6.0.3",
-		"deasync": "0.1.26",
 		"dotenv": "^10.0.0",
 		"eslint": "^8.32.0",
 		"eslint-config-wpcalypso": "5.0.0",

